### DISCUSSION TITLE
Do we need to filter the CNV data as well for each cohort while using the cohort analysis to compare cohorts?

### DISCUSSION THREAD

**Discussion by satyakam001** [PoisonAlien/maftools#935] [General]
Thank you for this wonderful tool! This is more of a question than an issue/error.

While running cohort analysis, for 3-4 different cohorts, let's say A, B, C and D, I have subsetted the maf into 4 different mafs. Now when adding custom CNV data to the mix, can I simply use the overall CNV data (which has all cohorts' sample names in it) or will I have to subset the CNV data for each category/cohort and store them?

Hence when I read maf for cohort A, will it be 

```r
read.maf("A.maf", cnTable=overall_CNV_data_table)
```

Or will it be 

```r
read.maf("A.maf", cnTable=CNV_data_table_filtered_for_cohort_A)
```

The thing is that some samples will be present in multiple cohorts. For instance, samples 12, 14 and 20 can be found in cohorts A and C, samples 13, 16, 18 can be found in B and A. Hence, will the results between using overall an CNV table and filtered CNV table be different? If yes, which one is the correct way to go?

Thank you!

**Comment by PoisonAlien** [comment]
Hi,
Thank you for the question and I am glad you find the package useful.
If you are woking with the cohorts independently, I would recommend the option B. First option will read in CNV from all the samples (it won't subset for the samples from maf).

**Comment by satyakam001** [comment]
Thank you so much! Yes that makes sense; 
I also wanted to know if I used subsetMAF function to subset a maf according to levels of a categorical variable for cohort comparison, how different would those mafs be than the mafs I read separately after I subset the .txt/.tsv original file using just the subset/filter options and convert them individually to mafs?
Basically, is subsetMAF doing the same thing as filtering txt file, converting it into maf and reading it back in?

Thank you!

**Reply by PoisonAlien** [reply]
Yeah pretty much the same. However it won't write to text file and read it again. It's all done in-memory.

**Reply by satyakam001** [reply]
Great, Thank you!
I did have one more question- While doing a cohort analysis using the CNV data, the results generated rank the genes' significance on the basis of their presence/absence. I was wondering if it is possible to determine the significance of the genes based on their status as gain of function/loss of function, or Amplification/Deletion/ShallowAmp/DeepDel?

We have a set of genes that are not significantly differentially mutated between different cohorts, but do show a major change in terms of gain/loss between the cohorts. It would be great if that data could be interpreted that way since it will have a significant biological implication.
